nucypher/acumen/perception.py

```diff
@@ -171,5 +171,14 @@ def abridged_state_details(state):
     def mark_as(self, label: Exception, node: "Teacher"):
         self._marked[label].append(node)
 
-        if self._nodes.get(node):
-            del self._nodes[node]
+        checksum_address = node.checksum_address
```

Comment:
I think this uncovers a more general problem. If the checksum address can't be relied upon to be unique, why are we using it as an index? Shouldn't we rather use, say, public key instead? This way the node can verify that it owns the corresponding private key by signing its metadata, and two nodes with the same private key may be safely treated as one node.

Reply:
> If the checksum address can't be relied upon to be unique

What makes you say that they are not unique?

Reply:
Isn't that what is being handled here? That we get a `node` which has the same checksum address as some known node, but is actually a different node.

Reply:
I've gotta pass this question off to @jMyles, since it's hos original authorship.  I have to agree with the second half of the comment, I don't really think it's possible for this to be changed?

To be clear the checksum address here is the ethereum address, certainly unique enough.

Reply:
It seems that I had an incorrect understanding, judging by the code we do verify that the node owns the private key for the checksum address it claims to have. Then I don't really understand what kind of situation this check covers.

Reply:
Me neither, I relocated it as part of this PR but I'm inclined to remove it and see if the tests pass normally.